internal/storage/v2/grpc/tracereader.go

```diff
@@ -48,11 +51,25 @@ func (tr *TraceReader) GetServices(ctx context.Context) ([]string, error) {
 	return resp.Services, nil
 }
 
-func (*TraceReader) GetOperations(
-	context.Context,
-	tracestore.OperationQueryParams,
+func (tr *TraceReader) GetOperations(
+	ctx context.Context,
+	params tracestore.OperationQueryParams,
 ) ([]tracestore.Operation, error) {
-	panic("not implemented")
+	resp, err := tr.client.GetOperations(ctx, &storage.GetOperationsRequest{
+		Service:  params.ServiceName,
+		SpanKind: params.SpanKind,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("%w: %w", errFailedToGetOperations, err)
```

Comment:
I would in-line the first string

Reply:
@yurishkuro this allows us to nicely test the error branch without string comparison - let me know your thoughts

Reply:
Generally I am not a fan of using production code in tests (even if it's a constant). But more importantly, the convention for `%w` is to be used in `"message: %w"`, i.e. where the error is logged only once. It creates a nested structure that works with things like `errors.Is`, etc. I don't know how this works if you're logging the error twice - all you really need for the first argument is `%s`.